### DISCUSSION TITLE
The perfect emulation machine

### DISCUSSION THREAD

**Discussion by IDmedia** [emmercm/igir#1340] [General]
For years I've wanted to build myself a Batocera/Retrobat machine. Datahording is one thing, but it has a rather low WIF (wife acceptance factor) when roms and emulators don't "just work" and are ready to be played like a game console. Now with the awesome igir tool, I feel it's feasible to actually create a solution that would automatically import/sort games without a lot of work.

I started looking into 1G1R as I find it both confusing and a waste of space having up to 4 copies of a single game (I still want to know that I have any game, but region doesn't matter that much to me). However, after starting to do some testruns I see that there's quite a difference between compability between USA, EUR, WORLD and JAP roms (maybe especially for newer systems).

Are there anyone else in the same situation? You want to have one of each rom, while trying to keep compability high? What strategy have you decided on using? My wife came up with the idea to drop 1G1R and rather keep a copy of all USA, WORLD, EUR and JAP (only exclusives) in each directory. I like that approach but, it would make browsing games a bit more confusing, but searching for games favorites would work great. The only problem is that while igir would (probably) be able to make filtered lists of each region, creating a JAP exclusives only would probably be hard. Also not sure if you would save enough space to make this approach worth it.

Would love to hear others solution to keep as many roms as possible with as high compability as possible without having to store the whole romset...

**Edit:** While I don't like Launchbox, this plugin sounds awesome: https://forums.launchbox-app.com/files/file/234-archive-cache-manager/
This would make it possible to store roms as compressed files, potentionally saving quite a bit of space while still being able to easily play them. Would love to see something like this for Batocera or RetroBat, but as far as I can tell you have events for when a game starts/stops, but you can't intercept/block the event and handle it your way instead (pulling file, extracting it, caching, playing).

**Comment by emmercm** [comment]
I'm curious about "compatibility" here, do you mean emulator support or performance is different for the different releases?

Or are the releases functionally different, as in maybe they shouldn't be clones of each other? Or is the reliability of the information in DATs wrong? Neither would surprise me since DATs are a crowd-sourced effort.

**Reply by IDmedia** [reply]
Emulator support. One example is LittleBigPlanet for PS3, the USA version doesn't work correctly and crashes upon boot (even if updated) while the European version works without any issues out of the box. It's a emulator/game related issue and not an issue with the DATs or Igir. :)

I'm currently thinking it would make the most sense to collect USA and Europe while only keeping Japan exclusives, but I guess that wouldn't be easy to do. If not, keeping all three of them would still probably save some disk space (would be able to skip Spain and other regions), but not sure if it's worth it compared to just collect the whole romset then. While Igir is able to filter by region, I'm not quite sure on how to only get the Japan exclusives yet. I've also noticed that Igir doesn't always pick the right region (I'll make a proper bug report for that later).

**Reply by emmercm** [reply]
That's interesting about the emulator compatibility. Maybe there's a way to marry the emulator's compatibility list with the Redump DAT, maybe with something similar to Retool.
